### PR TITLE
Add save book file

### PR DESCRIPTION
**Pull request by estebanmual**
## Learning objectives
- Store data in files.
- Run a program using the command line.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -137,11 +126,65 @@ def list_rental
...
+  def load_people(filename = 'people.json')
+    JSON.parse(File.read(filename)).each do |person|
+      @people << if person['type'] == 'teacher'
+                   Teacher.new(person['specialization'], person['age'], person['name'])
+                 else
+                   Student.new(person['age'], Classroom.new(person['classroom']), person['parent_permission'],
+                               person['name'])
+                 end
+    end
+  end
```
Comment (andersonlebon) [inline app.rb]: image


- [x] Kindly note that one of the requirements says that On startup of your app load all data from files, and handle the case when one or more of the files are missing (the app should work and show no errors). Kindly fix the error that occurs when one of the `json` files misses. Hint :point_down:

```rb 
def load_people(filename = 'people.json')
    people = File.exist?(filename) ? JSON.parse(File.read(filename)) : []
    people.each do |person|
      @people << if person['type'] == 'teacher'
                   Teacher.new(person['specialization'], person['age'], person['name'])
                 else
                   Student.new(person['age'], Classroom.new(person['classroom']), person['parent_permission'],
                               person['name'])
                 end
    end
  end
```

File: app.rb [hunk 2]
```diff
@@ -137,11 +126,65 @@ def list_rental
...
+    save_books
+    save_people
+    save_rentals
+  end
+
+  def load_books(filename = 'books.json')
+    JSON.parse(File.read(filename)).each do |book|
+      @books << Book.new(book['title'], book['author'])
+    end
+  end
```
Comment (andersonlebon) [inline app.rb]: image


- [x] Kindly note that one of the requirements says that On startup of your app load all data from files, and handle the case when one or more of the files are missing (the app should work and show no errors). Kindly fix the error that occurs when one of the `json` files misses. Hint :point_down:

```rb 
  def load_books(filename = 'books.json')
    books = File.exist?(filename) ? JSON.parse(File.read(filename)) : []
    books.each do |book|
      @books << Book.new(book['title'], book['author'])
    end
  end
```

File: app.rb [hunk 3]
```diff
@@ -137,11 +126,65 @@ def list_rental
...
+
+  def load_rentals(filename = 'rentals.json')
+    JSON.parse(File.read(filename)).each do |rental|
+      person = @people.find { |individual| individual.name = rental['person']['name'] }
+      book = @books.find { |tome| tome.title = rental['book']['title'] }
+      @rentals << Rental.new(rental['date'], book, person)
+    end
+  end
+end
```
Comment (andersonlebon) [inline app.rb]: image

- [x] Kindly note that one of the requirements says that On startup of your app load all data from files, and handle the case when one or more of the files are missing (the app should work and show no errors). Kindly fix the error that occurs when one of the `json` files misses. Hint :point_down:

```rb 
  def load_rentals(filename = 'rentals.json')
    rentals = File.exist?(filename) ? JSON.parse(File.read(filename)) : []
    rentals.each do |rental|
      person = @people.find { |individual| individual.name = rental['person']['name'] }
      book = @books.find { |tome| tome.title = rental['book']['title'] }
      @rentals << Rental.new(rental['date'], book, person)
    end
  end
end
```

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @estebanmual 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi team😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done working on the required changes from the previous review.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**